app/validate_txs.go

```diff
@@ -31,101 +64,106 @@ func separateTxs(_ client.TxConfig, rawTxs [][]byte) ([][]byte, []*tx.BlobTx) {
 	return normalTxs, blobTxs
 }
 
-// FilterTxs applies the antehandler to all proposed transactions and removes
-// transactions that return an error.
-//
-// Side-effect: arranges all normal transactions before all blob transactions.
-func FilterTxs(logger log.Logger, ctx sdk.Context, handler sdk.AnteHandler, txConfig client.TxConfig, txs [][]byte) [][]byte {
-	normalTxs, blobTxs := separateTxs(txConfig, txs)
-	normalTxs, ctx = filterStdTxs(logger, txConfig.TxDecoder(), ctx, handler, normalTxs)
-	blobTxs, _ = filterBlobTxs(logger, txConfig.TxDecoder(), ctx, handler, blobTxs)
-	return append(normalTxs, encodeBlobTxs(blobTxs)...)
-}
+func (fb *FilteredBuilder) Fill(txs [][]byte) [][]byte {
+	normalTxs, blobTxs := separateTxs(fb.txConfig, txs)
+
+	var (
+		nonPFBMessageCount = 0
+		pfbMessageCount    = 0
+		dec                = fb.txConfig.TxDecoder()
+		n                  = 0
+		m                  = 0
+	)
 
-// filterStdTxs applies the provided antehandler to each transaction and removes
-// transactions that return an error. Panics are caught by the checkTxValidity
-// function used to apply the ante handler.
-func filterStdTxs(logger log.Logger, dec sdk.TxDecoder, ctx sdk.Context, handler sdk.AnteHandler, txs [][]byte) ([][]byte, sdk.Context) {
-	n := 0
-	nonPFBMessageCount := 0
-	for _, tx := range txs {
+	for _, tx := range normalTxs {
 		sdkTx, err := dec(tx)
 		if err != nil {
-			logger.Error("decoding already checked transaction", "tx", tmbytes.HexBytes(coretypes.Tx(tx).Hash()), "error", err)
+			fb.logger.Error("decoding already checked transaction", "tx", tmbytes.HexBytes(coretypes.Tx(tx).Hash()), "error", err)
+			continue
+		}
+
+		if !fb.builder.AppendTx(tx) {
 			continue
 		}
 
 		// Set the tx size on the context before calling the AnteHandler
-		ctx = ctx.WithTxBytes(tx)
+		fb.ctx = fb.ctx.WithTxBytes(tx)
 
 		msgTypes := msgTypes(sdkTx)
 		if nonPFBMessageCount+len(sdkTx.GetMsgs()) > appconsts.MaxNonPFBMessages {
-			logger.Debug("skipping tx because the max non PFB message count was reached", "tx", tmbytes.HexBytes(coretypes.Tx(tx).Hash()))
+			fb.logger.Debug("skipping tx because the max non PFB message count was reached", "tx", tmbytes.HexBytes(coretypes.Tx(tx).Hash()))
 			continue
 		}
```

Comment:
[blocking][question] the builder already appended the tx on line 85 but here we're skipping running the handler on it and we haven't reverted it from the builder. I would expect this to call

```go
		if nonPFBMessageCount+len(sdkTx.GetMsgs()) > appconsts.MaxNonPFBMessages {
			fb.logger.Debug("skipping tx because the max non PFB message count was reached", "tx", tmbytes.HexBytes(coretypes.Tx(tx).Hash()))
			err = fb.builder.RevertLastTx()
			if err != nil {
				fb.logger.Error("reverting last transaction", "error", err)
			}
			continue
		}
```

Reply:
Alternatively this bug isn't present in the blob txs section below because appending happened after this message count check so we could move it down like so:

```go
		msgTypes := msgTypes(sdkTx)
		if nonPFBMessageCount+len(sdkTx.GetMsgs()) > appconsts.MaxNonPFBMessages {
			logger.Debug("skipping tx because the max non PFB message count was reached", "tx", tmbytes.HexBytes(coretypes.Tx(tx).Hash()))
			continue
		}


		if !fb.builder.AppendTx(tx) {
			continue
		}
```